### ISSUE TITLE
Various packages required for installation on Debian bullseye

### ISSUE THREAD

**Issue by failbit** [maxileith/homebridge-appletv-enhanced#6]
PIP installation was failing because of a needed package "libffi-dev". After installing it, the build continues.

Also the package "python3-dev".

also: build-essential libssl-dev cargo pkg-config

rust is too old in Debian bullseye to compile the cryptography pip dependency. so "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh" is needed aswell.

**Comment by failbit** [comment]
Also the package "python3-dev".

**Comment by failbit** [comment]
also: build-essential libssl-dev cargo pkg-config

**Comment by failbit** [comment]
rust is too old in Debian bullseye to compile the cryptography pip dependency. so "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh" is needed swell.

**Comment by maxileith** [comment]
I did not experience any of these problems on Raspberry PI OS. However, it could be the case that I had them already installed. But should be everything in the dependencies that is required when installing python3 and python3-venv.

I'm honestly a bit confused about this issue.

**Comment by failbit** [comment]
It's the Homebridge Raspberry Pi Image (https://homebridge.io/raspberry-pi-image) based on Raspbian Lite (maybe that makes a difference)

**Comment by maxileith** [comment]
Okay, actually I am using exactly that for my productive system ... but I have probably installed all these packages at some point before which is why I did not stumble across that ... seems like I have to test it on a fresh install and check / document what packages need to be installed.

Thank you for your valuable feedback.

**Comment by maxileith** [comment]
@failbit I have just flashed an SD Card with the latest Homebridge Image for Raspberry Pi. I am running a Raspberry PI 3B.

I updated the system with `sudo apt update` and `sudo apt upgrade -y`. I have then upgraded the Node Version to 20.9.0 via the `hb-service update-node` command. Then updated the Homebridge UI and Homebridge itself ... besides the upgrade to bookworm, everything should be uptodate.

I have then installed the plugin, configured it via Homebridge UI and restarted the homebridge.

The plugin complained that `python3-venv` is not available ... as it should be ... I installed it via `apt` and restarted again ...

After that the plugin created the venv, updated pip, installed the requirements, discovered my Apple TV, initilized the pairing request and after enter the PIN it showed up in the Home app.

So basically as it should be. I could not reproduce the problems you were mentioning.

Any thoughts on that?

**Comment by failbit** [comment]
Okay good to know. The only difference then is, that my Homebridge Base-Installation ist really old. I did 2-3 Distribution-Updates, so I guess that's where my missing packages came from. If it works with fresh installation of newer Homebridge-Image, all is good. Can be closed here.
